app/models/manageiq/providers/vmware/cloud_manager/refresh_parser.rb

```diff
@@ -58,6 +60,26 @@ def get_vms
     process_collection(@inv[:vms], :vms) { |vm| parse_vm(vm) }
   end
 
+  def get_private_images
+    @inv[:orgs].each do |org|
+      org.catalogs.each do |catalog|
+        get_images(catalog.catalog_items) unless catalog.is_published
```

Comment:
Oh no I would do exactly what you have here, but instead of calling `get_images` [here](https://github.com/ManageIQ/manageiq/pull/10136/files/ce6b73bec7ce322926acfc8467218b396cec7cd5#diff-36aecabb26f28b8069f3038e0e185874R66) just do `@inv[:catalogs] += catalog.catalog_items` then call `get_images(@inv[:catalogs])` at the end